File: Programmers/[12973] 짝지어 제거하기/pinkishincoloragain/12973.js
function solution(s) {
  let stack = [];
  for (let i = 0; i < s.length; i++) {
    if (stack[stack.length - 1] !== s[i]) {
      stack.push(s[i]);
    } else {
      stack.pop();
    }
  }
  return stack.length === 0 ? 1 : 0;
}

File: Programmers/[12973] 짝지어 제거하기/SUbbb/12973.java
import java.util.*;

class Solution {
    public int solution(String s) {
        Stack<Character> stack = new Stack<>();
        int len = s.length();
        
        for (int i = 0; i < len; i++) {
            char ch = s.charAt(i);
            char stackCh = ' ';
            
            if (!stack.isEmpty()) stackCh = stack.peek();
            
            if (ch == stackCh) stack.pop();
            else stack.push(ch);
        }
        
        return stack.isEmpty() ? 1 : 0;
    }
}